.gitlab-ci.yml

```diff
@@ -5,6 +5,16 @@
   tags:
     - docker
 
+cppcheck:
+  image: prplfoundationinc/prplmesh-builder:alpine-3.11.3
+  stage: build
+  script:
+    - tools/docker/static-analysis/cppcheck.sh common
```

Comment:
Since eventually, we will run it on all folders - why not just make cppcheck.sh accept no argument and internally only run the check on common for now?

Reply:
Also, why do we run it only on common? why not run it on all folders and be done with it? Can you explain the reason in the commit message?

Reply:
I wanted to keep the usage of `cppcheck.sh` flexible, to be able to call it on any folder namely (you can even give options to `cppcheck.sh`, they will be passed to cppcheck).

In CI we can only run it on `common` for now because the other folders still have many issues of severity 'warning' and 'error' to solve first. The job is made to fail if any of those issues remain.

Reply:
So you can add an optional parameter which is the directory to check. If none given, default should be to check all.